### PR TITLE
DOC Autogeneration of getting started code

### PR DESCRIPTION
**Pull request by merveenoyan**
This PR solves #126.

I also made a fixture out of metadata creation from config file so that it can be used repeatedly. Let me know if this is not good.

### CODE REVIEW

**Comment by merveenoyan** [comment]
Model card created from this branch, the code is automatically generated: https://huggingface.co/merve/test-code-generation

File: skops/card/_model_card.py [hunk 1]
```diff
@@ -380,6 +380,13 @@ def _generate_card(self) -> ModelCard:
         # add evaluation results
 
         template_sections = copy.deepcopy(self._template_sections)
+        if self.metadata:
+            model_file = self.metadata.to_dict().get("model_file", None)
```
Comment (BenjaminBossan) [inline skops/card/_model_card.py:384]: nit: not required

```suggestion
            model_file = self.metadata.to_dict().get("model_file")
```

**Review by BenjaminBossan — APPROVED** [review]
Overall LGTM.

One minor thing I wonder: Do we want to assume that users will always load the metadata from config or are there good reasons not to? If there are reasons not to, perhaps we should keep an example of manually adding the `get_started_code`.

File: skops/card/_model_card.py (hunk 1 above)
Comment (merveenoyan) [reply skops/card/_model_card.py:384]: They can already add the section if they want to with `.add()`, it looks a bit ugly IMO that's why I removed it.

**Comment by merveenoyan** [comment]
@BenjaminBossan I think it looks a bit ugly and our example already includes metadata from config file so that's why I wanted to remove it (it will be a duplication if we add that too).

**Comment by BenjaminBossan** [comment]
Yes, I'm aware that in the example, it doesn't make sense. What I meant is to document it some other place so that users have a chance of discovering it (as is, the only way to discover the feature is by reading the template).

If we can assume, however, that the vast majority of users create the section automatically, then maybe it's not necessary.

**Comment by merveenoyan** [comment]
@adrinjalali what do you suggest?

**Comment by adrinjalali** [comment]
I think since w/o the config file people won't really would know what to do with the model file, it's safe to assume the config file is there, especially for an example in the skops library, which generates that file.

File: skops/card/_model_card.py [hunk 2]
```diff
@@ -380,6 +380,13 @@ def _generate_card(self) -> ModelCard:
         # add evaluation results
 
         template_sections = copy.deepcopy(self._template_sections)
+        if self.metadata:
+            model_file = self.metadata.to_dict().get("model_file")
+            if model_file:
+                template_sections["get_started_code"] = (
+                    f"import pickle\nwith open({model_file}, 'rb') as file:\n    clf ="
+                    " pickle.load(file)"
```
Comment (adrinjalali) [inline skops/card/_model_card.py:388]: I think we can also add a `predict` example here, with the data available in the config file.

**Review by adrinjalali — COMMENTED** [review]
otherwise LGTM

File: skops/card/_model_card.py (hunk 2 above)
Comment (merveenoyan) [reply skops/card/_model_card.py]: Addressed. new version of card: https://huggingface.co/merve/test-code-generation

**Review by adrinjalali — COMMENTED** [review]
Thanks, this LGTM. I'd suggest at the same time we also remove the `<details>` tag around it, this is something I'd say should be visible to users.

**Comment by merveenoyan** [comment]
@adrinjalali done, new model card: https://huggingface.co/merve/test-code-generation

**Review by adrinjalali — APPROVED** [review]
That looks gorgeous, thanks for giving an example one on the Hug @merveenoyan , it really does help with the review.

**Pull request merged**